crates/sui-json-rpc/src/transaction_execution_api.rs

```diff
@@ -103,12 +99,16 @@ impl WriteApiServer for TransactionExecutionApi {
         match response {
             ExecuteTransactionResponse::EffectsCert(cert) => {
                 let (effects, is_executed_locally) = *cert;
+                let module_cache = self
```

Comment:
this is where we used to cache the module cache, which was not updated across epochs.
Now we use the module cache coming from the state which should always be the correct one

Reply:
I think there is a potential race here. The cert can be executed in epoch N, but by the time we get the response, the local fullnode may be at epoch N+1, so we may be using the epoch store from the new epoch.

However, this should be fine, because all we want is the module cache, and it shouldn't matter which epoch that comes from.

@longbowlu does that sound right to you?

Reply:
If all we use the modules from the module cache for is type resolution, we should also be protected by the fact that layouts for existing types must be preserved across upgrades (including for the framework).